motor/frameworks/asyncio.py

```diff
@@ -198,59 +154,54 @@ def settimeout(self, timeout):
     @asyncio_motor_sock_method
     @asyncio.coroutine
     def connect(self):
-        protocol_factory = lambda: self
-
-        # TODO: will call getaddrinfo again.
-        host, port = self.options.address
-        self._transport, protocol = yield from self.loop.create_connection(
-            protocol_factory, host, port,
-            ssl=self.ctx)
+        is_unix_socket = (self.options.family == getattr(socket,
+                                                         'AF_UNIX', None))
+        if self.options.use_ssl:
+            # TODO: cache at Pool level.
+            ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+            if self.options.certfile is not None:
+                ctx.load_cert_chain(self.options.certfile,
+                                    self.options.keyfile)
+            if self.options.ca_certs is not None:
+                ctx.load_verify_locations(self.options.ca_certs)
+            if self.options.cert_reqs is not None:
+                ctx.verify_mode = self.options.cert_reqs
+                if ctx.verify_mode in (ssl.CERT_OPTIONAL, ssl.CERT_REQUIRED):
+                    ctx.check_hostname = True
+        else:
+            ctx = None
+
+        if is_unix_socket:
+            path = self.options.address[0]
+            reader, writer = yield from asyncio.open_unix_connection(
+                path, loop=self.loop, ssl=ctx)
```

Comment:
Not sure. We have no test for it but I've passed ssl parameter as `open_unix_connection accepts it.
Frankly speaking I don't know does Mongo support SSL-ed unix sockets at all.